src/site/content/en/blog/async-clipboard/index.md

```diff
@@ -0,0 +1,413 @@
+---
+title: Unblocking clipboard access
+subhead: Safer, unblocked clipboard access for text and images
+authors:
+  - jasonmiller
+  - thomassteiner
+description: Async Clipboard API simplifies permissions-friendly copy and paste.
+date: 2020-07-31
+updated: 2020-07-31
+tags:
+  - blog
+  - capabilities
+  - input
+  - clipboard
+hero: hero.jpg
+alt: Clipboard with shopping list
+---
+
+Over the past few years, browsers have used
+`[document.execCommand()](https://developers.google.com/web/updates/2015/04/cut-and-copy-commands)`
+for clipboard interactions. Though widely supported, this method of cutting and
+pasting came at a cost: clipboard access was synchronous, and could only read
+and write to the DOM.
+
+That's fine for small bits of text, but there are many cases where blocking the
+page for clipboard transfer is a poor experience. Time consuming sanitization or
+image decoding might be needed before content can be safely pasted. The browser
+may need to load or inline linked resources from a pasted document. That would
+block the page while waiting on the disk or network. Imagine adding permissions
+into the mix, requiring that the browser block the page while requesting
+clipboard access. At the same time, the permissions put in place around
+`document.execCommand()` for clipboard interaction are loosely defined and vary
+between browsers.
+
+The
+[Async Clipboard API](https://www.w3.org/TR/clipboard-apis/#async-clipboard-api)
+addresses these issues, providing a well-defined permissions model that doesn't
+block the page. Safari recently announced  [support for it in version
+13.1](https://webkit.org/blog/10855/). With that, major browsers have a basic
+level of support in place. As of this writing, Firefox only supports text and
+image support is limited to pngs in some browsers. If you're interested in using
+the API,
+[consult a browser support table](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard#Browser_compatibility)
+before proceeding.
+
+This article explains how to write both text and images to and read them from
+the clipboard. This article does not cover [Feature Policy for the
+clipboard](https://www.chromestatus.com/features/5767075295395840) which landed
+in Chrome 85.
+
+{% Aside %}
+The Async Clipboard API is limited to handling text and images. Chrome 84
+introduces an experimental feature that allows the clipboard to handle any
+arbitrary data type. For more information, see Cut and paste non-standard data
```

Comment:
Is there a link missing?

Reply:
Apparently to content I intended to write. This rewrite was stalled for more than a month. It's hard to remember everything you intended to do.